packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx

```diff
@@ -0,0 +1,236 @@
+}
+
+const DocumentActions = ({ actions }: DocumentActionsProps) => {
+  const { formatMessage } = useIntl();
+
+  const [primaryAction, secondaryAction, ...restActions] = actions.filter(
+    (action) => action.position !== 'header'
+  );
+
+  if (!primaryAction) {
+    return null;
+  }
+
+  const isMenuDisabled = restActions.every((action) => action.disabled);
+
+  return (
+    <Flex direction="column" gap={2} alignItems="stretch" width="100%">
+      <Flex gap={2}>
+        <Button
+          flex={1}
+          startIcon={primaryAction.icon}
+          disabled={primaryAction.disabled}
+          onClick={primaryAction.onClick}
+          justifyContent="center"
+        >
+          {primaryAction.label}
+        </Button>
+        {restActions.length > 0 ? (
+          <Menu.Root>
+            <Menu.Trigger
+              disabled={isMenuDisabled}
+              size="S"
+              endIcon={null}
+              paddingTop="7px"
+              paddingLeft="9px"
+              paddingRight="9px"
+              variant="secondary"
+            >
+              <More aria-hidden focusable={false} />
+              <VisuallyHidden as="span">
+                {formatMessage({
+                  id: 'content-manager.containers.edit.panels.default.more-actions',
+                  defaultMessage: 'More actions',
+                })}
+              </VisuallyHidden>
+            </Menu.Trigger>
+          </Menu.Root>
+        ) : null}
+      </Flex>
+      {secondaryAction ? (
+        <Button
+          startIcon={secondaryAction.icon}
+          disabled={secondaryAction.disabled}
+          onClick={secondaryAction.onClick}
+          justifyContent="center"
+          variant="secondary"
+        >
+          {secondaryAction.label}
+        </Button>
+      ) : null}
+    </Flex>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * DocumentActionComponents
+ * -----------------------------------------------------------------------------------------------*/
+
+const PublishAction: DocumentActionComponent = ({ activeTab, id, model, collectionType }) => {
+  const navigate = useNavigate();
+  const { formatMessage } = useIntl();
+  const canPublish = useDocumentRBAC('PublishAction', (state) => state.canPublish);
+  const { publish } = useDocumentActions();
+  const modified = useForm('UpdateAction', ({ modified }) => modified);
+  const isSubmitting = useForm('PublishAction', ({ isSubmitting }) => isSubmitting);
+
+  return {
+    disabled: !canPublish || isSubmitting || activeTab === 'published' || modified,
+    label: formatMessage({
+      id: 'app.utils.publish',
+      defaultMessage: 'Publish',
+    }),
+    onClick: async () => {
+      if (id) {
+        const res = await publish({
+          collectionType,
+          model,
+          id,
+        });
+
+        if ('data' in res) {
+          navigate({ search: `?status=published` });
```

Comment:
Shouldn't we make sure we keep the locale here?

Reply:
I've not implemented i18n for this branch yet, i'll do all that at that point :)